### DISCUSSION TITLE
Routing hooks

### DISCUSSION THREAD

**Discussion by tigawanna** [rakkasjs/rakkasjs#91] [Q&A] [answered]
Are there any routing hooks like useRouter for nextjs , Nextjs has the ability to check if the new page is loading in order to show a progress bar at the top before navigating the page or instead of showing a loading screen

**Comment by cyco130** [comment, marked as answer]
Sure, try `useLocation` which returns an object with a `current` and a `pending` property which are `URL` objects. `pending` is only defined during page transitions.

It's documented [here](https://rakkasjs.org/guide/client-side-navigation).
